packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts

```diff
@@ -1085,5 +1090,68 @@ import  { Value4 } from 'default_and_named_import';
         },
       ],
     },
+    {
+      // https://github.com/typescript-eslint/typescript-eslint/issues/2775
+      code: `
+        import Default, * as Rest from 'module';
+        const a: Rest.A = '';
+      `,
+      options: [{ prefer: 'type-imports' }],
+      // eslint-disable-next-line @typescript-eslint/internal/plugin-test-formatting
+      output: noFormat`
+        import type * as Rest from 'module';
+import Default from 'module';
+        const a: Rest.A = '';
+      `,
```

Comment:
I'd consider just dedenting the entire block to 0 (both code and output) so it matches instead of disabling the lint rule
```suggestion
      output: noFormat`
import type * as Rest from 'module';
import Default from 'module';
const a: Rest.A = '';
      `,
```